fabric-item-group-api-v1/src/main/java/net/fabricmc/fabric/impl/itemgroup/ItemGroupHelper.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.itemgroup;
+
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+import org.jetbrains.annotations.ApiStatus;
+
+import net.minecraft.item.ItemGroup;
+import net.minecraft.item.ItemGroups;
+
+import net.fabricmc.fabric.api.itemgroup.v1.FabricItemGroup;
+import net.fabricmc.fabric.mixin.itemgroup.ItemGroupAccessor;
+import net.fabricmc.fabric.mixin.itemgroup.ItemGroupsAccessor;
+
+@ApiStatus.Internal
+public final class ItemGroupHelper {
+	private ItemGroupHelper() {
+	}
+
+	public static void appendItemGroup(FabricItemGroup itemGroup) {
+		for (ItemGroup existingGroup : ItemGroups.GROUPS) {
+			if (existingGroup.getId().equals(itemGroup.getId())) {
+				throw new IllegalStateException("Duplicate item group: " + itemGroup.getId());
+			}
+		}
+
+		final int index = ItemGroups.GROUPS.length;
+		final ItemGroup[] itemGroups = Stream.concat(Arrays.stream(ItemGroups.GROUPS), Stream.of(itemGroup))
+				.toArray(ItemGroup[]::new);
```

Comment:
Don't we have the apache commons lib that has a util method for adding an element to an array? Doesn't really matter though tbh

Reply:
ArrayUtils seems to do the job, thanks 👍